Automattic/jetpack
# Site Migration: Add site migration endpoint for access to migration key

Related to https://github.com/Automattic/wp-calypso/issues/87393

## Proposed changes:

* Add a `/wpcom/v2/sites/${site}/site-migration` endpoint to sites with the Migrate Guru plugin, from which we can fetch the migration key on the front end.

Let me know if there's a better approach I'm missing. I haven't been able to get this to show the new endpoint for a WoA site using the Jetpack Beta plugin, or by copying the file directly to the WoA site, but it seems to work on Simple sites (although those don't have the plugin, so the migration key will be an empty string.)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply this patch to your sandbox and sandbox `public-api.wordpress.com`
* From the dev console access `/wpcom/v2/sites/%s/site-migration` for your Simple test site
* Confirm you see an array with a `migration_key => ''` -- the value will be empty because the class doesn't exist on Simple sites.

<img width="715" alt="Screenshot 2024-03-06 at 4 09 13 PM" src="https://github.com/Automattic/jetpack/assets/2124984/d1431c55-8105-4667-ac25-6783844a7af2">

* I'm not sure how to get it working on Atomic sites; I'm running into authorization errors when I try to access the endpoint via the dev console or even internally. 🤔
